::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
: Copyright (c) 2018 Enea AB and others.
:
: All rights reserved. This program and the accompanying materials
: are made available under the terms of the Apache License, Version 2.0
: which accompanies this distribution, and is available at
: http://www.apache.org/licenses/LICENSE-2.0
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Mon, 21 Aug 2017 20:42:00 +0200
Subject: [PATCH] seedng: module: Add AArch64 repo

salt custom py module seedng.py should use custom repo arg
"-R linux.enea.com/saltstack" on AArch64 nodes.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 .../0105-seedng-module-Add-AArch64-repo.patch      | 22 ++++++++++++++++++++++
 mcp/patches/patches.list                           |  1 +
 2 files changed, 23 insertions(+)
 create mode 100644 mcp/patches/0105-seedng-module-Add-AArch64-repo.patch

diff --git a/mcp/patches/0105-seedng-module-Add-AArch64-repo.patch b/mcp/patches/0105-seedng-module-Add-AArch64-repo.patch
new file mode 100644
index 0000000..d40e63b
--- /dev/null
+++ b/mcp/patches/0105-seedng-module-Add-AArch64-repo.patch
@@ -0,0 +1,22 @@
+From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
+Date: Mon, 21 Aug 2017 02:03:01 +0200
+Subject: [PATCH] seedng: module: Add AArch64 repo
+
+salt custom py module seedng.py should use custom repo arg
+"-R linux.enea.com/saltstack" on AArch64 nodes.
+
+Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
+---
+
+diff --git a/_modules/seedng.py b/_modules/seedng.py
+--- a/_modules/seedng.py
++++ b/_modules/seedng.py
+@@ -257,6 +257,7 @@
+              or salt.syspaths.BOOTSTRAP)
+     # Exec the chroot command
+-    arg = 'stable {0}'.format('.'.join(salt.version.__version__.split('.')[:2]))
++    arg = '-R linux.enea.com/saltstack ' if os.uname()[-1] == 'aarch64' else ''
++    arg += 'stable {0}'.format('.'.join(salt.version.__version__.split('.')[:2]))
+     cmd = 'if type salt-minion; then exit 0; '
+     cmd += 'else sh {0} -c /tmp {1}; fi'.format(
+         os.path.join(tmppath, 'bootstrap-salt.sh'), arg)
diff --git a/mcp/patches/patches.list b/mcp/patches/patches.list
index 9ce1cc9..efcf366 100644
--- a/mcp/patches/patches.list
+++ b/mcp/patches/patches.list
@@ -19,3 +19,4 @@
 /usr/share/salt-formulas/env: 0101-maas-Add-curtin_userdata_arm64_generic_xenial.patch
 /usr/share/salt-formulas/env: 0103-virtng-module-Extend-libvirt_domain.patch
 /usr/share/salt-formulas/env: 0104-salt-control-virt-Extend-libvirt_domain.patch
+/usr/share/salt-formulas/env: 0105-seedng-module-Add-AArch64-repo.patch
